packages/sign-client/src/controllers/engine.ts

```diff
@@ -884,6 +988,15 @@ export class Engine extends IEngine {
       throw new Error(`Could not find pending auth request with id ${id}`);
     }
 
+    const transportType =
+      pendingRequest.transportType || this.isLinkModeEnabled(pendingRequest.requester.metadata)
+        ? "link-mode"
+        : "relay";
+
+    if (transportType === "relay") {
+      await this.confirmOnlineStateOrThrow();
+    }
+
```

Comment:
Shouldn't we change `encodeOpts` at line 1005 to TYPE_1 or TYPE_2 depending on the transport type?

Reply:
not sure TBH, are the other SDKs using type_2 in that case?

Reply:
i think its type_1 but for link_mode it will be encoded in `BASE64URL`